atintegrators/BeamLoadingCavityPass.c

```diff
@@ -72,16 +74,20 @@ void BeamLoadingCavityPass(double *r_in,int num_particles,int nbunch,
     int *pslice;
     double *kz;
     double freqres = rffreq/(1-tan(vgenk[1])/(2*qfactor));
-    double vgen = vgenk[0];
-    double psi = vgenk[1];
     
     for(i=0;i<nbunch;i++){
         tot_current += bunch_currents[i];
     }
 
-    /*Track RF cavity is always done. */
-    trackRFCavity(r_in,le,vgen/energy,rffreq,harmn,tlag,-psi,nturn,circumference/C0,num_particles);
+    if(cavitymode==2){
```

Comment:
Here I would not change anything and integrate in trackRFCavity that if vgen=0 the loop over particle is skipped

Reply:
done